src/NuGet.Core/NuGet.Packaging/Signing/Authoring/X509SignatureProvider.cs

```diff
@@ -141,7 +141,7 @@ private static PrimarySignature CreatePrimarySignature(CmsSigner cmsSigner, Sign
 
             try
             {
-                cms.ComputeSignature(cmsSigner);
+                cms.ComputeSignature(cmsSigner, false); // silent is false to ensure PIN prompts appear if CNG/CAPI requires it
```

Comment:
@PatoBeltran as you're working on .net core 2.1 signing stuff, please don't forget this change too.

Reply:
p.s. we'll make sure to replicate this change! Can you tell me more about why this is needed? I'm still fairly new to this work and the codebase, myself :)

Reply:
The issue is that if you use a smart card/fob that requires an interactive prompt for a password, you need to make sure it appears.

I'm not quite sure how to automate a test for it, but in my case, I have a certificate on a fob that uses the SafeNet Authentication Client. When a cert operation is invoked, it prompts a dialog for the password to the certificate.

Without my change, the silent is true, meaning it won't prompt and the operation will fail.

Reply:
I think @onovotny has an equivalent PR for the feature branch. 

https://github.com/NuGet/NuGet.Client/pull/3069

Reply:
Anyone who uses yubikey for their certificate will need the false flag set as well.